### ISSUE TITLE
File path has dangerous characters

### ISSUE THREAD

**Issue by mrtaxi** [Tobi823/ffupdater#365]
Stacktrace:
```
java.lang.IllegalArgumentException: File path has dangerous characters: /storage/emulated/0/Android/data/de.marmaro.krt.ffupdater/files/Download/org_mozilla_fenix_2023-03-06 04:01.apk
	at de.marmaro.krt.ffupdater.installer.impl.RootInstaller.fileIsSafeOrThrow(RootInstaller.kt:29)
	at de.marmaro.krt.ffupdater.installer.impl.RootInstaller.executeInstallerSpecificLogic(RootInstaller.kt:21)
	at de.marmaro.krt.ffupdater.installer.impl.AbstractAppInstaller.install2Internal(AbstractAppInstaller.kt:43)
	at de.marmaro.krt.ffupdater.installer.impl.AbstractAppInstaller.access$install2Internal(AbstractAppInstaller.kt:15)
	at de.marmaro.krt.ffupdater.installer.impl.AbstractAppInstaller$startInstallation$2.invokeSuspend(AbstractAppInstaller.kt:21)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@4a9d540, Dispatchers.Main]
Timestamp: 2023-03-06T17:18:30.481

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.10 (136) release |
| Device | Pixel (OnePlus7, OnePlus7, msmnile) |
| Manufacturer | Google (Google) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 11 (SDK: 30) |
| OS | rd-build-75, OnePlus, release-keys, 1634800224000 |

**Comment by Tobi823** [comment]
I found the bug:

String::replace("old", "new") will not accept Regex. I have to call "old".toRegex() first.

**Comment by Tobi823** [comment]
Will be fixed in 77.7.11
